src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1150,107 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Returns the collectible by the address and token id.
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @returns a collectible
+   */
+  findCollectibleByAddressAndTokenId(
+    address: string,
+    tokenId: string,
+  ): { collectible: Collectible; index: number } | null {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address.toLowerCase() === address.toLowerCase() &&
+        collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return null;
+    }
+
+    return { collectible: collectibles[index], index };
+  }
+
+  /**
+   * Update collectible data.
+   *
+   * @param collectible - Collectible object to update.
+   */
+  updateCollectibleByAddressAndTokenId(collectible: Collectible) {
```

Comment:
If this takes a collectible and only a collectible, do you need the `ByAddressAndTokenId` at the end? What about just `updateCollectible`?

Reply:
Actually, now that I realize it, I wonder if it would make more sense for this to take a different set of arguments. If you say:

``` typescript
updateCollectible(collectible)
```

What would you expect it to do? It feels incomplete that this just takes one argument. "Update collectible with what?" would be my question. So I think either this should take two arguments, `collectible: Collectible` and `updates: Partial<Collectible>`, e.g.

``` typescript
updateCollectible(existingCollectible, { transactionId: 1 })
```

or it should take three arguments, `address`, `tokenId`, and `updates`, in which case you can keep the existing name:

``` typescript
updateCollectibleByAddressAndTokenId(address, tokenId, { transactionId: 1 })
```

What do you think?